mz_bokeh_package/utilities/environment.py

```diff
@@ -27,75 +27,71 @@ def get_environment() -> str:
-        If the environment variable 'ENVIRONMENT' is set to 'dev' the environment variable 'GRAPHQL_API_HOST' should
-        be set to the desired url, including 'https://'. If 'GRAPHQL_API_HOST' is not set in the development
-        environment, the default staging url will be returned.
+        """Returns the url of the GraphQL API Server. The host should be set in the environment variable
+        'GRAPHQL_API_HOST'.
 
         Returns:
             the full URL of the GraphQL API server
-        """
-
-        env = cls.get_environment()
-
-        if env == 'staging':
-            host = 'https://api-staging.materials.zone/graphql'
-        elif env == 'production':
-            host = 'https://api.materials.zone/graphql'
-        elif env == 'dev':
-            host = os.getenv('GRAPHQL_API_HOST', 'https://api-staging.materials.zone/graphql')
 
+        Raises:
+            ValueError: When the environment variable `GRAPHQL_API_HOST` is not set.
+        """
+        host = Environment._get_host_from_env_var('GRAPHQL_API_HOST')
         return f"{host}"
 
     @classmethod
     def get_parser_service_url(cls, endpoint: str) -> str:
-        """receives an endpoint of a Parser Service request and converts it to a request url based on the environment.
-        If the environment variable 'ENVIRONMENT' is set to 'dev' the environment variable 'PARSER_SERVICE_HOST' should
-        be set to the desired url, including 'https://'. If 'PARSER_SERVICE_HOST' is not set in the development
-        environment, the default staging url will be returned.
+        """Converts an endpoint of the Parser Service to a request url. The host should be set in the environment
+        variable 'PARSER_SERVICE_HOST'.
 
         Args:
             endpoint: the endpoint of the request
 
         Returns:
             the full URL of the request
+
+        Raises:
+            ValueError: When the environment variable `PARSER_SERVICE_HOST` is not set.
         """
 
-        env = cls.get_environment()
+        host = Environment._get_host_from_env_var('PARSER_SERVICE_HOST')
+        return f"{host}/{endpoint}"
 
-        if env == 'staging':
-            host = 'https://parser-service-staging.materials.zone/api/v1beta1'
-        elif env == 'production':
-            host = 'https://parser-service.materials.zone/api/v1beta1'
-        elif env == 'dev':
-            host = os.getenv('PARSER_SERVICE_HOST', 'https://parser-service-staging.materials.zone/api/v1beta1')
+    @classmethod
+    def get_webapp_host(cls) -> str:
+        """Get the web app host as set in the environment variable WEBAPP_HOST.
 
-        return f"{host}/{endpoint}"
+        Returns:
+            str: Web app host.
+
+        Raises:
+            ValueError: When the environment variable `WEBAPP_HOST` is not set.
+        """
+        return Environment._get_host_from_env_var('WEBAPP_HOST')
+
+    @classmethod
+    def _get_host_from_env_var(cls, env_var_name: str):
```

Comment:
The function would get any environment variable. The additional feature is that it raises an exception when it fails. Calling it `_getenv_or_raise_value_error` is bulky, but it would also solve the riddle of the calling functions having (correctly, I find) the `Raises` section in their docstring despite them not raising anything but just letting an exception go past them.

Reply:
@captKuk I agree with this comment. I renamed the method as you suggested (I don't think the name is too bulky for a private method). I also removed the "Raised" parts from the docstrings and add a comment about the `ValueError` to the description.

Reply:
@oriyudilevich , I didn't think of that before, but there is a function/property in the `os` module, that - implicitly - does precisely what the function `_getenv_or_raise_value_error` does:
```
>>> os.environ["a_known_variable"]
the_content_of_the_variable
>>> os.environ["an_undefined_variable"]
Traceback
...
KeyError: 'an_undefined_variable'
```
 Well, it raises a key error. But that doesn't make a difference for our purposes...

Reply:
@captKuk I tried to make the change you suggested, but decided to revert after trying to run the ingester with the new error message, cause there is value in having the clearer message `The WEBAPP_HOST environment variable is not set`. It could save the (new) developer a few minutes of trying to figure out what the issue is. I did change the exception to a `KeyError`, cause that makes more sense than `ValueError`.